### ISSUE TITLE
ttnn.group_norm fused kernel silently produces wrong output for non-tile-aligned H*W

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-metal#50682]
### Observed

- `ttnn.group_norm` (fused kernel) **silently produces numerically wrong output** when the per-sample flattened height `H*W` is not a multiple of the tile height (32). No exception is raised — the op runs and returns a plausible-looking tensor.
- The fused kernel computes each group's mean/variance over the **tile-padded** `H*W` rows (rounded up to the next multiple of 32) instead of the real `H*W` rows, so the padding rows contaminate the statistics. A correct affine-free group_norm normalizes every group to **mean 0, std 1**; on non-tile-aligned `H*W` the per-group mean drifts from 0 and std from 1.
- The shape is **not rejected**: a logically non-tile-aligned `H*W` is silently accepted and executed.
- Important measurement note: **PCC vs the correct reference stays ≈ 1.0**, because the contamination is a per-group *affine* (scale + shift) transform, and Pearson correlation is invariant to affine transforms. PCC therefore does **not** detect this bug — the per-group **mean/std** and the **absolute error** do.

### Error (no exception — silently wrong output)

`GroupNorm(num_groups=32, C=1024)` on Blackhole, vs `torch.nn.functional.group_norm` (tt-metal's own group_norm test tolerance is `atol=0.065`):

```
TILE-ALIGNED H*W (control - correct):
  H*W=256 (pad  0.0%) | per-group mean=0.0257 (want ~0)  std=0.9857 (want ~1)  maxErr_vs_torch=0.059  PCC=0.99999

NON-TILE-ALIGNED H*W (BUG - fused kernel reduces over tile-padding rows):
  H*W=200 (pad 10.7%) | per-group mean=0.1701 (want ~0)  std=0.8969 (want ~1)  maxErr_vs_torch=0.361  PCC=0.99999
  H*W= 40 (pad 37.5%) | per-group mean=0.5485 (want ~0)  std=0.8313 (want ~1)  maxErr_vs_torch=0.884  PCC=0.99993
```

- The error **scales monotonically with the padding fraction** (0% → within tolerance; 37.5% → `maxErr 0.884`), and the wrong output matches a "reduce-over-tile-padded-rows" reference to within bf16 noise — confirming the fused kernel reduces over the tile-padding.
- Every non-tile-aligned case exceeds tt-metal's own group_norm test tolerance (`atol=0.065`) by **4–14×**, while PCC stays ≈ 1.0.

### Expected

- For non-tile-aligned `H*W`, `ttnn.group_norm` should produce a correct result — per-group **mean 0, std 1**, matching `torch.nn.functional.group_norm` within tolerance — or reject the input up front, rather than silently returning a numerically wrong tensor.

### Steps (exact commands)

Save as `repro_group_norm_non_tile_aligned.py` and run with the tt-metal Python env:

```bash
python repro_group_norm_non_tile_aligned.py
```

```python
# ttnn.group_norm fused kernel is silently wrong for non-tile-aligned H*W.
# Correct (affine-free) group_norm => per-group mean=0, std=1, matches torch within bf16 tol.
import math, torch, ttnn

def pcc(a, b):
    a, b = a.flatten().float(), b.flatten().float()
    a, b = a - a.mean(), b - b.mean()
    d = a.norm() * b.norm()
    return float((a @ b) / d) if d > 0 else float("nan")

def run(dev, C, HW, G):
    torch.manual_seed(0)
    x = torch.rand((1, C, 1, HW), dtype=torch.bfloat16)                       # [N, C, H=1, W=H*W]
    ref = torch.nn.functional.group_norm(x.float(), G).permute(0, 2, 3, 1).reshape(1, 1, HW, C)
    xin = x.permute(0, 2, 3, 1).reshape(1, 1, HW, C)                          # ttnn layout [N, 1, H*W, C]
    tt = ttnn.from_torch(xin, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                         device=dev, memory_config=ttnn.DRAM_MEMORY_CONFIG)
    o = ttnn.to_torch(ttnn.from_device(
        ttnn.group_norm(tt, num_groups=G, inplace=False, memory_config=ttnn.DRAM_MEMORY_CONFIG)
    )).float().reshape(1, 1, HW, C)
    og = o.reshape(1, HW, G, C // G)
    padded = math.ceil(HW / 32) * 32
    print(f"  H*W={HW:>3} (pad {100*(padded-HW)/padded:4.1f}%) | "
          f"per-group mean={og.mean(dim=(1,3)).abs().max():.4f} (want ~0)  "
          f"std={og.std(dim=(1,3), unbiased=False).mean():.4f} (want ~1)  "
          f"maxErr_vs_torch={(o-ref).abs().max():.3f}  PCC={pcc(ref,o):.5f}")

dev = ttnn.open_device(device_id=0)
print("GroupNorm(num_groups=32, C=1024).  tt-metal group_norm test tolerance atol=0.065\n")
print("TILE-ALIGNED H*W (control - correct):")
run(dev, 1024, 256, 32)
print("\nNON-TILE-ALIGNED H*W (BUG - fused kernel reduces over tile-padding rows):")
run(dev, 1024, 200, 32)
run(dev, 1024, 40, 32)
ttnn.close_device(dev)
```

Control (tile-aligned `H*W`) matches torch within tolerance (mean ~0, std ~1). The non-tile-aligned cases return wrong statistics (see output above).

### Software Versions

- Ubuntu 24.04.4 LTS
- tt-metal @ `13adda80c119631d18b0bc06163416ba148c25ab`
- Surfaced from tt-mlir @ `260d4c49` (tenstorrent/tt-mlir#8935) / tt-xla XTTS-v2 conditioning encoder, whose `GroupNorm(32, 1024)` has a mel length ~269 (non-tile-aligned).

**Comment by ashokkumarkannan1** [comment]
https://github.com/tenstorrent/tt-metal/pull/51159 fixes the issue
